src/bin/bench.rs

```diff
@@ -184,18 +190,35 @@ fn perform_tree_operations(tree: Tree, config: Config) {
         (Op::Delete, config.delete),
         (Op::Cas, config.cas),
     ];
-
-    // TODO update to handle division errors
-    let ops_per_thread = config.num_operations / config.num_threads;
+    let ops_per_thread = config.num_operations / config.num_threads; // TODO update to handle division errors
     let mut threads = Vec::new();
+    let ops_per_second = AtomicUsize::new(0);
+    let histo = Arc::new(Histo::default());
+
     let tree = Arc::new(tree);
     let ops = Arc::new(ops);
+    let ops_per_second = Arc::new(ops_per_second);
+
+    // thread which spits out bench-related results every 1 second
+    {
+        let ops_per_second = ops_per_second.clone();
+        thread::spawn(move || {
+            loop {
+                thread::sleep(Duration::seconds(1).to_std().unwrap());
```

Comment:
`thread::sleep(Duration::from_secs(1))`

Reply:
Thanks for commenting 👍!

I had that before, but now since I'm using the `Duration` struct from `time` crate I had a naming conflict. I decided to pick one to keep the usage consistent and it this case it was the `Duration` struct from `time`.

Reply:
Lol. Looks like `time` is no longer maintained (https://github.com/rust-lang-deprecated/time#notes). Maybe I switch to `chronos`. I wish that `cargo` shows the READMEs...